boggle-backend/boggle-lib/src/main/java/com/boggle/Tree.java

```diff
@@ -0,0 +1,73 @@
+package com.boggle;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Tree {
+    private Node root;
+
+    public Tree(){
+        this.root = new Node();
+    }
+
+    public boolean isPrefix(String word){
+        Node currentNode = root;
+        for (int i = 1; i < word.length(); i++){
+            char currentLetter = word.charAt(i);
+            currentNode.getChildren();
+            if (currentNode.childHasLetter(currentLetter) == true){
+                continue;
+            }
+            else{
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public void createTree(String word){
+        Node currentNode = root;
+        Map<Character, Node> characterToNode = new HashMap<>();
+        for (int i = 1; i < word.length(); i++){
+            characterToNode = currentNode.getChildren();
+            if (characterToNode.containsKey(word.charAt(i))){
+                currentNode = characterToNode.get(word.charAt(i));
+            }
+            else{
+                currentNode.createNode(word.charAt(i));
+                currentNode = characterToNode.get(word.charAt(i));
+            }
+        }
+
+    }
+
+    public List<String> wordsPerTree(char startLetter){
+        String path = "";
+        List<String> wordsPerTree = new ArrayList<>();
+        Node currentNode = root;
+        Map<Character, Node> characterToNode = new HashMap<>();
+        characterToNode = currentNode.getChildren();
+        path += startLetter;
+        createWordsPerTree(path, wordsPerTree, currentNode);
+        return wordsPerTree;
+    }
+
+    public void createWordsPerTree(String path, List<String> words, Node currentNode){
```

Comment:
i think this should be a private function cause people should only call wordsPerTree